src/overlays/actors/ovl_Boss_03/z_boss_03.c

```diff
@@ -2458,90 +2459,92 @@ void Boss03_SeaweedUpdate(Actor* thisx, GlobalContext* globalCtx) {
     Boss03* this = THIS;
     s16 i;
     s16 pad;
-    s16 phi_s0;
-    u8 phi_s3;
+    s16 maxBendSpeed;
+    u8 flag;
     Player* player = GET_PLAYER(globalCtx);
-    f32 temp_f0;
-    f32 temp_f14;
+    f32 distanceBetweenSeaweedAndDisturbance;
+    f32 disturbanceFactor;
     f32 xDiff;
     f32 yDiff;
     f32 zDiff;
 
-    phi_s3 = 0;
+    flag = 0;
     this->unk_240++;
 
-    for (i = 0; i < 6; i++) {
-        xDiff = player->actor.world.pos.x - this->unk_2DC[i].x;
-        yDiff = player->actor.world.pos.y - this->unk_2DC[i].y;
-        zDiff = player->actor.world.pos.z - this->unk_2DC[i].z;
-        temp_f0 = sqrtf(SQ(xDiff) + SQ(yDiff) + SQ(zDiff));
-        temp_f14 = player->actor.speedXZ * 3.0f + 70.0f;
+    for (i = 0; i < ARRAY_COUNT(this->seaweedSegmentPositions); i++) {
+        xDiff = player->actor.world.pos.x - this->seaweedSegmentPositions[i].x;
+        yDiff = player->actor.world.pos.y - this->seaweedSegmentPositions[i].y;
+        zDiff = player->actor.world.pos.z - this->seaweedSegmentPositions[i].z;
+        distanceBetweenSeaweedAndDisturbance = sqrtf(SQ(xDiff) + SQ(yDiff) + SQ(zDiff));
+        disturbanceFactor = player->actor.speedXZ * 3.0f + 70.0f;
 
-        // Player is on the floor
+        // Player is standing on ground
         if (player->actor.bgCheckFlags & 1) {
-            phi_s0 = 0;
+            maxBendSpeed = 0;
         } else {
-            phi_s0 = player->actor.speedXZ * 16.0f;
-            if (phi_s0 > 0x1000) {
-                phi_s0 = 0x1000;
-            } else if (phi_s0 < 0x100) {
-                phi_s0 = 0x100;
+            maxBendSpeed = player->actor.speedXZ * 16.0f;
+            if (maxBendSpeed > 0x1000) {
+                maxBendSpeed = 0x1000;
+            } else if (maxBendSpeed < 0x100) {
+                maxBendSpeed = 0x100;
             }
         }
 
-        if (temp_f0 < temp_f14) {
-            Math_ApproachS(&this->morphTable[i].x, (temp_f14 - temp_f0) * 200.0f, 0xA, phi_s0);
-            if (phi_s0 != 0) {
-                phi_s3 |= 1;
+        if (distanceBetweenSeaweedAndDisturbance < disturbanceFactor) {
+            Math_ApproachS(&this->morphTable[i].x, (disturbanceFactor - distanceBetweenSeaweedAndDisturbance) * 200.0f,
+                           0xA, maxBendSpeed);
+            if (maxBendSpeed != 0) {
+                flag |= 1;
             }
         }
     }
 
-    if (phi_s3 & 1) {
+    if (flag & 1) {
```

Comment:
If you really wanted to you could make defines for Player and for George here, but probably just a comment somewhere is plenty.